packages/suite/src/views/wallet/transactions/TransactionList/useFetchTransactions.ts

```diff
@@ -0,0 +1,223 @@
+import { useCallback, useEffect, useMemo, useState } from 'react';
+
+import { getTxsPerPage } from '@suite-common/suite-utils';
+import {
+    getIsPhishingTransaction,
+    selectNetworkTokenDefinitions,
+} from '@suite-common/token-definitions';
+import { DiscoveryStatus } from '@suite-common/wallet-constants';
+import {
+    fetchAllTransactionsForAccountThunk,
+    fetchTransactionsPageThunk,
+    selectAccountTotalTransactions,
+    selectAccountTransactionsWithNulls,
+    selectIsLoadingAccountTransactions,
+} from '@suite-common/wallet-core';
+import { getSynchronize } from '@trezor/utils';
+
+import { useDiscovery, useDispatch, useSelector } from 'src/hooks/suite';
+import { Account, WalletAccountTransaction } from 'src/types/wallet';
+
+import { shouldAttemptToLoadNextPageForVisibleTransactions } from './transaction-fetch-utils';
+
+const getPaging = (network: Account['networkType'], txFetched: number, txTotal: number) => {
+    const perPage = getTxsPerPage(network);
+    // There is no total in XRP, so always presume there could be one more tx and calculate page count accordingly
+    const totalItems = network === 'ripple' ? txFetched + 1 : txTotal;
+    const pagesTotal = Math.ceil(totalItems / perPage);
+    // Consider incomplete pages unfetched unless fetched tx count equals total
+    const page = txFetched === totalItems ? pagesTotal : Math.floor(txFetched / perPage);
+
+    return { page, pagesTotal, perPage };
+};
+
+export const useFetchTransactions = (
```

Comment:
this is made by @marekrjpolak I like the API, it is not neceasarily needed, but the API is nice.